src/index.ts

```diff
@@ -64,30 +42,58 @@ export class Client {
     return pack.version
   }
 
-  /**
-   * config is the (public) threads config.
-   */
-  public readonly config: Config
-
   /**
    * Client creates a new gRPC client instance.
    * @param config A set of configuration settings to control the client.
    */
-  constructor(config: Config | BaseConfig = {}) {
-    if (config instanceof Config) {
-      this.config = config
-    } else {
-      this.config = new Config(config.host, config.transport)
-    }
-    grpc.setDefaultTransport(this.config.transport)
+  constructor(public config: Config = defaultConfig) {
```

Comment:
Removed the checks here so it'll just take whatever it gets and treat it as a Config. This makes it easier to use in js-textile.